### PR TITLE
Add ios build for travis

### PR DESCRIPTION
**Pull request by halx99**

### CODE REVIEW

**Comment by kcat** [comment]
For some reason it seems to not be recognizing NEON extensions, but does get SSE and `cpuid`. Doesn't iphone use ARM? Why would it be recognizing x86 things?

**Comment by halx99** [comment]
because previous commit, I use iphonesimulator as sysroot, and after I change to ```iphoneos```, just failed as I said: https://github.com/kcat/openal-soft/issues/406

**Comment by halx99** [comment]
I have test this https://github.com/leetal/ios-cmake toolchan, it works.
But I meet crash issue when I try to integrate latest cocos2d-x v4.
and I realized the openal latest reqiured c++14, but the cocos2dx v4 default c++ standard is 11
may be the different compiler cause the crash, I will continue test later

**Comment by halx99** [comment]
I have tested, openal-soft works fine on both ios simulator and devices, so I think this PR could be merge

File: CMakeLists.txt [hunk 1]
```diff
@@ -134,6 +134,7 @@ check_type_size("long" SIZEOF_LONG)
 # Require C++14
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_EXTENSIONS FALSE)
```
Comment (kcat) [inline CMakeLists.txt:137]: I don't think I like this. It inhibits optional but beneficial features that could be useful for the generated code.

Comment (halx99) [reply CMakeLists.txt]: agree with you, removed

**Pull request merged**